src/darwin/Framework/CHIP/MTRBaseDevice.mm

```diff
@@ -2497,6 +2497,24 @@ ConcreteEventPath path(static_cast<chip::EndpointId>([endpointID unsignedShortVa
     return [[MTREventPath alloc] initWithPath:path];
 }
 
+- (BOOL)isEqualToEventPath:(MTREventPath *)eventPath
+{
+    return [self isEqualToClusterPath:eventPath] && [_event isEqualToNumber:eventPath.event];
+}
+
+- (BOOL)isEqual:(id)object
+{
+    if (![object isKindOfClass:[self class]]) {
```

Comment:
The basic bug is that `MTRClusterPath` should be checking for `[object class] == [self class]`, not `isKindOfClass:` since instances of different sub-classes should never be equal. The sub-class implementations can then be `return [super isEqual:obejct] && /* test the sub-class-specific fields */`.

As this is written, an `[aClusterPath isEqual:someEventPath]` will still incorrectly compare only the cluster path parts.

Reply:
I guess that depends on the expected semantics of isEqual.  If someone hands me an `MTRClusterPath*` that happens to be an event path under the hood, and I do `[myClusterPath isEqual:givenClusterPath]`, should it test equal?

But I guess in that situation we'd want the same answer as `[givenClusterPath isEqual:myClusterPath]`, which presumably should be false?

Reply:
Fixed.